### PR TITLE
[webpack] update typings for webpack 2

### PR DESCRIPTION
**Pull request by bumbleblym**
- remove features removed in webpack 2
- update `CommonsChunkPlugin` constructor type
- add `Watching`/`MultiCompiler`/`MultiWatching` to `webpack` function return type
- change `Plugin` `apply` param from `Webpack` to `Compiler`
- remove `Webpack`, `Optimize`, `Dependencies`, and plugin static types

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

File: webpack-stream/index.d.ts [hunk 1]
```diff
@@ -1,48 +1,26 @@
-// Type definitions for webpack-stream v3.2.0
+// Type definitions for webpack-stream 3.2
```
Comment (bcherny) [inline webpack-stream/index.d.ts]: Is this the format types-publisher expects?

**Comment by bcherny** [comment]
TBH This is too big to review with any sort of confidence. The changes look reasonable, and this was obviously a lot of work. 👍 if you've tested this thoroughly.

**Comment by bcherny** [comment]
:+1:

File: webpack-stream/index.d.ts (hunk 1 above)
Comment (bumbleblym) [inline webpack-stream/index.d.ts]: yep; https://github.com/Microsoft/types-publisher/blob/6308963c0c0c7ec32f05f3ee20f2174fa2343d26/src/lib/header.ts#L7

**Pull request merged**